### PR TITLE
Add basic alerts checkbox to job

### PR DESCRIPTION
**Pull request by tedpearson**
This allows creation of three types of alerts automatically with just a checkbox. (error, runtime, and rekick) Emails and webhooks are associated with user groups, and can be configured in hydra-main application.conf.

### CODE REVIEW

File: hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertRunner.java [hunk 1]
```diff
@@ -346,19 +346,16 @@ private void sendEmailAlert(AbstractJobAlert jobAlert,
...
         String description = jobAlert.description;
-        boolean hasDescription = LessStrings.isNotEmpty(description);
-        String subject;
-        if (hasDescription) {
-            subject = reason + ' ' + description.split("\n")[0];
-        } else {
-            subject = String.format("%s %s - %s - %s", reason, jobAlert.getTypeString(),
-                                    JobAlertRunner.getClusterHead(), errors.keySet());
+        boolean blankDescription = StringUtils.isBlank(description);
+        String shortDescription = description.split("\n")[0];
```
Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertRunner.java]: Should guard against `description` being null

File: hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java [hunk 2]
```diff
@@ -258,7 +260,9 @@ private static void startWebInterface(Spawn spawn) throws Exception {
                                                       required = true) PermissionsManager permissionsManager,
                                @Nonnull @JsonProperty(value = "jobDefaults",
                                                       required = true) JobDefaults jobDefaults,
-                               @Bytes @JsonProperty(value = "datastoreCacheSize") long datastoreCacheSize) throws Exception {
+                               @Bytes @JsonProperty(value = "datastoreCacheSize") long datastoreCacheSize,
+                               @JsonProperty(value = "groupManager", required = true) GroupManager groupManager)
```
Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java]: @Nonnull?

File: hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
  */
 package com.addthis.hydra.job.alert;
 
+import java.util.Set;
+import java.util.UUID;
```
Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java]: Unused import

File: hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java [hunk 4]
```diff
@@ -37,13 +49,34 @@
...
+        this.scheduleAlertScan(scheduledExecutorService);
+    }
 
     public JobAlertManagerImpl(Spawn spawn, ScheduledExecutorService scheduledExecutorService) {
         this.spawnDataStore = spawn.getSpawnDataStore();
-        this.jobAlertRunner = new JobAlertRunner(spawn, areAlertsEnabled());
-        scheduleAlertScan(scheduledExecutorService);
+        this.jobAlertRunner = new JobAlertRunner(spawn, this.areAlertsEnabled());
+        this.groupManager = spawn.getGroupManager();
+        this.scheduleAlertScan(scheduledExecutorService);
```
Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java]: Better to call the constructor above.

File: hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java [hunk 5]
```diff
@@ -107,7 +140,169 @@ public String getAlert(String alertId) {
...
+     *
+     * @param job         the job being updated
+     * @param basicAlerts true if we want basic alerts
+     * @param basicPages  true if we want basic pages
+     */
+    @Override public void updateBasicAlerts(final IJob job, final boolean basicAlerts, final boolean basicPages) {
+        @Nullable final Group group = this.groupManager.getGroup(job.getGroup());
+        // don't try to create alerts if config is not set up for the user/group
+        if (group == null) {
+            log.info("group was null");
```
Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java]: Use log.warn, and include more info in the message such as job id and group name. Ideally the user gets some kind of feedback for this, but that may be a lot more code than we want to change now.

File: hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java (hunk 4 above)
Comment (tedpearson) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java]: Sorry, what?

Comment (yuesong) [inline hydra-main/src/main/java/com/addthis/hydra/job/alert/JobAlertManagerImpl.java]: I meant this constructor can call the other overloaded constructor - initialization code is the same with the exception of jobAlertRunner, which the other constructor takes as a parameter

**Comment by tedpearson** [comment]
There's still another bug I need to fix - if you remove rekick or runtime config from a job, corresponding alerts won't be removed.

**Pull request merged**